src/gazebo_mavlink_interface.cpp

```diff
@@ -1135,34 +1171,67 @@ void GazeboMavlinkInterface::pollForMAVLinkMessages()
     return;
   }
 
-  struct pollfd fds[1] = {};
-
-  if (use_tcp_) {
-    fds[0].fd = simulator_tcp_client_fd_;
-  } else {
-    fds[0].fd = simulator_socket_fd_;
-  }
-  fds[0].events = POLLIN;
-
-  bool received_actuator = false;
-
   do {
     int timeout_ms = (received_first_actuator_ && enable_lockstep_) ? 1000 : 0;
+    int ret = ::poll(&fds_[0], N_FDS, timeout_ms);
 
-    int ret = ::poll(&fds[0], 1, timeout_ms);
+    if (ret < 0) {
+      gzerr << "poll error: " << strerror(errno) << "\n";
+      return;
+    }
 
     if (ret == 0 && timeout_ms > 0) {
       gzerr << "poll timeout\n";
+      return;
     }
 
-    if (ret < 0) {
-      gzerr << "poll error: " << strerror(errno) << "\n";
-    }
+    for (int i = 0; i < N_FDS; i++) {
+      if(fds_[i].revents == 0)
+        continue;
 
-    if (fds[0].revents & POLLIN) {
+      if(fds_[i].revents != POLLIN)
+      {
+        gzerr << "invalid events at fd:" << i << "\n";
+        return;
+      }
+
+      if (i == LISTEN_FD)
+      { // accept call
+        do {
```

Comment:
Everything inside this if gets hard to understand. I suggest to move it to a method.

Reply:
Fixed it. See changes.